src/lightning/pytorch/callbacks/progress/rich_progress.py

```diff
@@ -632,6 +631,17 @@ def _reset_progress_bar_ids(self) -> None:
         self.test_progress_bar_id = None
         self.predict_progress_bar_id = None
 
+    @override
+    def get_metrics(
+        self, trainer: "pl.Trainer", pl_module: "pl.LightningModule"
+    ) -> dict[str, Union[int, str, float, dict[str, float]]]:
+        items = super().get_metrics(trainer, pl_module)
+        # convert all metrics to float before sending to rich
+        for k, v in items.items():
+            if isinstance(v, torch.Tensor):
+                items[k] = v.item()
+        return items
```

Comment:
```suggestion
        return apply_to_collection(items, torch.Tensor, lambda x: x.item())
```

just a suggestion. Requires import
```python
from lightning_utilities.core.apply_func import apply_to_collection
```